pkg/lambdalogger/logger.go

```
@@ -88,6 +88,10 @@ func ConfigureGlobal(
 		log.Panic("failed to build zap logger: " + err.Error())
 	}
 
+	// Sync the current global logger so we don't miss any log lines
+	if err := zap.L().Sync(); err != nil {
+		log.Panic("failed to flush zap logger: " + err.Error())
```

Should this be best effort ->  if syncing failed... move on?
Is panic helping here?

---

If syncing failed it probably means we cannot log anymore. The `zap` global logger as constructed here writes to `stdout`. Something must have gone completely wrong in the process state if we fail to `Sync()`. In addition this function does not return an error so we cannot handle this in any meaningful way.

---

In general I consider missing log lines a fatal error for our lambdas. Especially since we will be basing metrics on the `zap.Logger` as was decided.

---

TBH this should be changed to just `panic(err)` since we are probably missing the error again by feeding it to the logger that just failed to `Sync()` :)